### PR TITLE
Replace usage of M_PI in picongpu with Pi

### LINKED ISSUE #2485: Eliminate the usage of M_PI
Although widely used, M_PI and its derivative constants are not standard C++. It is also not ideal for template code, as type casts are normally required. 

Most of the code uses PI defined in physicalConstants.param (which also requires type casts, but this is not the point for this issue). However, some pieces still use M_PI, these should be changed to either PI or, in case it's not possible, with some new standard-conforming code.

### PR DESCRIPTION
**Pull request by sbastrakov**
This closes #2485.

### CODE REVIEW

File: include/picongpu/particles/bremsstrahlung/Bremsstrahlung.tpp [hunk 1]
```diff
@@ -149,7 +150,7 @@ float3_X Bremsstrahlung<T_IonSpecies, T_ElectronSpecies, T_PhotonSpecies>::scatt
     float_X sinTheta, cosTheta;
     math::sincos(theta, sinTheta, cosTheta);
 
-    const float_X phi = -float_X(M_PI) + float_X(2.0) * float_X(M_PI) * this->randomGen(acc);
+    const float_X phi = -math::Pi<float_X>::value + math::Pi<float_X>::doubleValue * this->randomGen(acc);
```
Comment (PrometheusPi) [inline include/picongpu/particles/bremsstrahlung/Bremsstrahlung.tpp]: CODING STYLE:
The template argument should be surrounded by spaces: `math::PI< float_X >::...`
see: https://github.com/ComputationalRadiationPhysics/contributing/blob/master/codingGuideLines/cpp.md#17-template-instantiations

File: include/picongpu/particles/bremsstrahlung/Bremsstrahlung.tpp [hunk 2]
```diff
@@ -211,13 +212,14 @@ unsigned int Bremsstrahlung<T_IonSpecies, T_ElectronSpecies, T_PhotonSpecies>::n
 
     /* electron deflection due to Rutherford scattering without modifying the electron
        energy based on radiation emission */
-    const float_X zMin = float_X(1.0) / (float_X(M_PI)*float_X(M_PI));
+    const float_X zMin = float_X(1.0) / (math::Pi<float_X>::value * math::Pi<float_X>::value);
```
Comment (PrometheusPi) [inline include/picongpu/particles/bremsstrahlung/Bremsstrahlung.tpp]: CODING STYLE:
same here - spaces around template argument

File: include/picongpu/particles/bremsstrahlung/Bremsstrahlung.tpp [hunk 3]
```diff
@@ -211,13 +212,14 @@ unsigned int Bremsstrahlung<T_IonSpecies, T_ElectronSpecies, T_PhotonSpecies>::n
...
     const float_X zMax = float_X(1.0) / (electron::MIN_THETA*electron::MIN_THETA);
     const float_X z = zMin + this->randomGen(acc) * (zMax - zMin);
     const float_X theta = math::rsqrt(z);
     const float_X targetZ = GetAtomicNumbers<T_IonSpecies>::type::numberOfProtons;
-    const float_X rutherfordCoeff = float_X(2.0) * ELECTRON_CHARGE*ELECTRON_CHARGE / (float_X(4.0) * float_X(M_PI) * EPS0) * targetZ / Ekin;
-    const float_X scaledDeflectionDCS = float_X(M_PI) * (zMax - zMin) * rutherfordCoeff*rutherfordCoeff;
+    const float_X rutherfordCoeff = float_X(2.0) * ELECTRON_CHARGE*ELECTRON_CHARGE /
+        (float_X(4.0) * math::Pi<float_X>::value * EPS0) * targetZ / Ekin;
```
Comment (PrometheusPi) [inline include/picongpu/particles/bremsstrahlung/Bremsstrahlung.tpp]: CODING STYLE:
same here - spaces around template argument

File: include/picongpu/particles/bremsstrahlung/Bremsstrahlung.tpp [hunk 4]
```diff
@@ -211,13 +212,14 @@ unsigned int Bremsstrahlung<T_IonSpecies, T_ElectronSpecies, T_PhotonSpecies>::n
...
+    const float_X scaledDeflectionDCS = math::Pi<float_X>::value * (zMax - zMin) * rutherfordCoeff*rutherfordCoeff;
```
Comment (PrometheusPi) [inline include/picongpu/particles/bremsstrahlung/Bremsstrahlung.tpp]: CODING STYLE:
same here - spaces around template argument

File: include/picongpu/particles/bremsstrahlung/PhotonEmissionAngle.hpp [hunk 5]
```diff
@@ -190,7 +190,7 @@ struct GetPhotonAngle
         minimum = boost::math::tools::brent_find_minima(
             aimForDelta,
             0.0,
-            M_PI,
+            pmacc::algorithms::math::Pi<float_64>::value,
```
Comment (PrometheusPi) [inline include/picongpu/particles/bremsstrahlung/PhotonEmissionAngle.hpp]: CODING STYLE:
same here - spaces around template argument

File: include/picongpu/particles/bremsstrahlung/ScaledSpectrum.tpp [hunk 6]
```diff
@@ -91,9 +92,10 @@ HDINLINE float_X LookupTableFunctor::operator()(const float_X Ekin, const float_
  */
 float_64 ScaledSpectrum::dcs(const float_64 Ekin, const float_64 kappa, const float_64 targetZ) const
 {
-    const float_64 bohrRadius = float_64(4.0) * M_PI * EPS0 * HBAR*HBAR / (ELECTRON_MASS * ELECTRON_CHARGE*ELECTRON_CHARGE);
-    const float_64 classicalElRadius = ELECTRON_CHARGE*ELECTRON_CHARGE / (float_64(4.0) * M_PI * EPS0 * ELECTRON_MASS * SPEED_OF_LIGHT*SPEED_OF_LIGHT);
-    const float_64 fineStructureConstant = ELECTRON_CHARGE*ELECTRON_CHARGE / (float_64(4.0) * M_PI * EPS0 * HBAR * SPEED_OF_LIGHT);
+    constexpr float_64 pi = pmacc::algorithms::math::Pi<float_64>::value;
```
Comment (PrometheusPi) [inline include/picongpu/particles/bremsstrahlung/ScaledSpectrum.tpp]: CODING STYLE:
same here - spaces around template argument 

:+1: great replacement here

File: include/picongpu/plugins/particleCalorimeter/ParticleCalorimeter.hpp [hunk 7]
```diff
@@ -285,8 +285,9 @@ class ParticleCalorimeter : public ISimulationPlugin
         this->allGPU_reduce = AllGPU_reduce(new pmacc::algorithm::mpi::Reduce<simDim>(zone_allGPUs));
 
         /* calculate rotated calorimeter frame from posYaw_deg and posPitch_deg */
-        const float_64 posYaw_rad = this->posYaw_deg * float_64(M_PI / 180.0);
-        const float_64 posPitch_rad = this->posPitch_deg * float_64(M_PI / 180.0);
+        constexpr float_64 radsInDegree = pmacc::algorithms::math::Pi<float_64>::value / float_64(180.0);
```
Comment (PrometheusPi) [inline include/picongpu/plugins/particleCalorimeter/ParticleCalorimeter.hpp]: CODING STYLE:
same here - spaces around template argument

File: include/picongpu/plugins/radiation/windowFunctions.hpp [hunk 8]
```diff
@@ -78,7 +80,7 @@ namespace picongpu
       {
         const float_X x = position_x - L_x*float_X(0.5);
         const float_X a = 0.08; /* ideal parameter: -43dB reduction */
-        const float_X cosinusValue = math::cos(M_PI*x/L_x);
+        const float_X cosinusValue = math::cos(pmacc::algorithms::math::Pi<float_X>::value*x/L_x);
```
Comment (PrometheusPi) [inline include/picongpu/plugins/radiation/windowFunctions.hpp]: CODING STYLE:
same here - spaces around template argument

File: include/picongpu/plugins/radiation/windowFunctions.hpp [hunk 9]
```diff
@@ -109,7 +111,7 @@ namespace picongpu
       {
         const float_X x = position_x - L_x*float_X(0.5);
         const float_X lambda = float_X(5.0)/L_x; /* larger is better, but too large means no data */
-        const float_X cosinusValue = math::cos(M_PI*x/L_x);
+        const float_X cosinusValue = math::cos(pmacc::algorithms::math::Pi<float_X>::value*x/L_x);
```
Comment (PrometheusPi) [inline include/picongpu/plugins/radiation/windowFunctions.hpp]: CODING STYLE:
same here - spaces around template argument

File: include/picongpu/simulation_defines/unitless/synchrotronPhotons.unitless [hunk 10]
```diff
@@ -35,7 +38,7 @@ constexpr float_64 SYNC_FUNCS_STEP_WIDTH =
 constexpr float_64 SYNC_FUNCS_F1_INTEGRAL_BOUND = 50.0;
 
 constexpr float_X SOFT_PHOTONS_CUTOFF_MOM = static_cast<float_X>(
-    HBAR * 2.0 * M_PI / SOFT_PHOTONS_CUTOFF_RATIO / DELTA_T / SPEED_OF_LIGHT);
+    HBAR * pmacc::algorithms::math::Pi<float_X>::doubleValue / SOFT_PHOTONS_CUTOFF_RATIO / DELTA_T / SPEED_OF_LIGHT);
```
Comment (PrometheusPi) [inline include/picongpu/simulation_defines/unitless/synchrotronPhotons.unitless]: CODING STYLE:
same here - spaces around template argument

File: include/picongpu/plugins/particleCalorimeter/ParticleCalorimeter.hpp (hunk 7 above)
Comment (ax3l) [inline include/picongpu/plugins/particleCalorimeter/ParticleCalorimeter.hpp]: maybe call this `radsToDegree` in order to be more explicit?

**Comment by ax3l** [comment]
@PrometheusPi I disagree with your annotations, the legacy code he changed uses old style around it. no need to make his line different without tooling support.

File: include/picongpu/particles/bremsstrahlung/Bremsstrahlung.tpp (hunk 1 above)
Comment (PrometheusPi) [reply include/picongpu/particles/bremsstrahlung/Bremsstrahlung.tpp]: see comment of @ax3l - this only applies to new files

Comment (ax3l) [reply include/picongpu/particles/bremsstrahlung/Bremsstrahlung.tpp]: if you like, you can "dismiss" your own review at the bottom

Comment (PrometheusPi) [reply include/picongpu/particles/bremsstrahlung/Bremsstrahlung.tpp]: Since this style issue was my only remark, I approved the pull request.

Comment (sbastrakov) [reply include/picongpu/particles/bremsstrahlung/Bremsstrahlung.tpp]: Yeah, I tried to match the brackets and spacing to the existing style in each file. However, ofc I might have still missed smth. And also a couple of files used mixed style, so I preferred a new style in those.

Comment (ax3l) [reply include/picongpu/particles/bremsstrahlung/Bremsstrahlung.tpp]: That's correct: try match existing style in old files (as long as reasonable) for small changes and go directly for new style in new files.

File: include/picongpu/plugins/particleCalorimeter/ParticleCalorimeter.hpp (hunk 7 above)
Comment (sbastrakov) [reply include/picongpu/plugins/particleCalorimeter/ParticleCalorimeter.hpp]: You mean `degreeToRad`? This is how many rads are there in one degree. Or maybe the name is bad and confused you.

**Pull request merged**